test/assertions.go

```diff
@@ -0,0 +1,19 @@
+package testutils
+
+import (
+	"context"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+// FailOnTimeout blocks until `ctx` is done or until specified timeout has elapsed.
+// In the latter case it calls require.FailNow(t, "test timed out").
+func FailOnTimeout(ctx context.Context, t *testing.T, timeout time.Duration) {
```

Comment:
this seems like a strange pattern to me, why does it not fail when the ctx is cancelled?
```
func FailOnTimeout(t *testing.T, f func(), timeout time.Duration)
```
would make more sense to me

Reply:
Agree, it goes against the common semantics. Cancellation of the context there means successful completion of the test when a long running (and potentially blocking) operation is executed in a goroutine, e.g. in `TestRangeBig`. I will redo using channels to avoid the confusion